sympy/codegen/array_utils.py

```diff
@@ -86,6 +86,15 @@ def __new__(cls, expr, *contraction_indices, **kwargs):
         obj._shape = shape
         return obj
 
+    # XXX: Maybe __mul__ and __rmul__ should have some type checking. These
+    # methods were added because Expr no longer accepts non-Expr in __mul__.
+
+    def __mul__(self, other):
+        return Mul(self, other)
+
+    def __rmul__(self, other):
+        return Mul(other, self)
```

Comment:
I would recommend:
```python
def __mul__(self, other):
    raise NotImplementedError("Product of N-dim arrays is not uniquely defined. Use another method.")
```
At least you can specify the error message.

In the future we may decide to convert these objects into an `Expr` (it's not so far-fetched, you may wish to put these expressions into mathematical formulae?).

Reply:
Thanks. I've added that but I included a special case for multiplying by 1 since presumably that's okay and that was the case in the test failure.